config/base-server.js

```diff
@@ -0,0 +1,30 @@
+const webpack = require('webpack');
+const utils = require('./utils');
+const base = require('./base');
+
+const devServer = utils.merge(base, {
+  target: 'node',
+  output: {
+    path: utils.resolve('dist/server'),
+    libraryTarget: 'commonjs',
+  },
+  plugins: [
+    new webpack.DefinePlugin({
+      CLIENT: false,
```

Comment:
We can use one or the other, `CLIENT: true | false` and get rid of `SERVER`.
`CLIENT` is enough to determine the context whether it is client or server, IMO.